container-di/src/main/scala/com/yahoo/container/di/ConfigRetriever.scala

```diff
@@ -43,16 +43,27 @@ final class ConfigRetriever(bootstrapKeys: Set[ConfigKeyT],
 
   private def getConfigsOptional(leastGeneration: Long): Option[ConfigSnapshot] = {
     val newestComponentGeneration = componentSubscriber.waitNextGeneration()
+    log.log(DEBUG, s"getConfigsOptional: new component generation: $newestComponentGeneration")
 
+    // leastGeneration is only used to ensure newer generation when the previous generation was invalidated due to an exception
     if (newestComponentGeneration < leastGeneration) {
       None
     } else if (bootstrapSubscriber.generation < newestComponentGeneration) {
       val newestBootstrapGeneration = bootstrapSubscriber.waitNextGeneration()
+      log.log(DEBUG, s"getConfigsOptional: new bootstrap generation: ${bootstrapSubscriber.generation}")
       bootstrapConfigIfChanged() orElse {
-        if (newestBootstrapGeneration == newestComponentGeneration) componentsConfigIfChanged()
-        else None
+        if (newestBootstrapGeneration == newestComponentGeneration){
+          log.log(DEBUG, s"Got new components configs with unchanged bootstrap configs.")
+          componentsConfigIfChanged()
+        }
+        else {
```

Comment:
Is it not better to use "} else {" as is done in the rest of the file.

Reply:
Fixed (force pushed into original commit)